spark3-extensions/src/main/antlr/org.apache.spark.sql.catalyst.parser.extensions/IcebergSqlExtensions.g4

```diff
@@ -76,6 +76,7 @@ callArgument
 
 expression
     : constant
+    | stringMap
```

Comment:
This is required to allow Map(string, string, *) expressions

Reply:
Looks like this is intended to mimic the [`map` function](https://spark.apache.org/docs/latest/api/sql/#map)?

If so, should this allow any constant for keys and values? I would expect something like this:

```
mapLiteral
    : MAP '(' pairs+=keyValue, (',' pairs+=keyValue)* ')'
    ;

keyValue
    : key=constant ',' value=constant
    ;
```

Reply:
Actually, we may want to remove `keyValue` and just use `constant` so that a missing value doesn't result in a parse error. Instead we can give a better error message in the AST builder.

Reply:
I considered this, but we really wanted to only support String maps and I was falling back on the expression level to tell us if it's a bad Map (unequal key values) but we can do that here if you like.

Reply:
Yeah, I think the list works. The main thing is whether we should allow passing non-string values.